src/samples/network/netctl.c

```diff
@@ -94,8 +97,58 @@ error_cb(void *data, const struct sol_netctl_service *service,
 }
 
 static void
+report_error(void *data, struct sol_netctl_service *service,
+    const char *error)
+{
+    int r;
+
+    printf("The agent action error is %s\n", error);
+    r = sol_netctl_report_error(service, SOL_NETCTL_AGENT_NORETRY);
+    printf("The agent report error return value is %d\n", r);
+}
+
+static void
+request_input(void *data, struct sol_netctl_service *service,
+    const struct sol_ptr_vector *vector)
```

Comment:
ahh... following my previous comment on "why do you free/destroy the vector", you pass the vector as a "const"... but you want the user to edit its fields... not good to use "const" here. Just remove const all over the place.